src/http/index.js

```diff
@@ -29,6 +29,24 @@ function hapiInfoToMultiaddr (info) {
   return toMultiaddr(uri)
 }
 
+async function serverCreator (serverAddrsArr, createServerFunc, hapiInfoToMultiaddr, ipfs) {
+  if (!serverAddrsArr.length) {
+    debug(Error('There are no addresses'))
```

Comment:
Please just `return []` here instead - it shouldn't be an error to not have any addresses to listen on.